enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,260 @@
+we will bring a new agent controller named `cluster-proxy-agent-controller` that
+qualifies the interfaces defined in addon-framework and automatically does the 
+following things upon discovering a new `ManagedClusterAddon` resource:
+
+1. Signing all the required X509 certificates and keys beforehand:
+   
+    i.   a CSR requesting server certificate which will be shared across the 
+         replicas of the proxy servers.
+   
+    ii.  a CSR requesting agent certificate which will be shared by the proxy 
+         agents.
+   
+    iii. a CSR requesting client certificate for logging onto in the tunnel.
+
+2. A configuration custom resource that contains the following information: 
+
+    - the expected replicas of proxy-servers.
+      
+    - the proxy protocol.
+      
+    - the server/agent ports
+    
+3. Installing proxy-server in the cluster namespace in the hub cluster.
+   
+4. Installing proxy-agent in an arbitrary namespace in the spoke cluster with 
+   bootstrap credentials configured.
+   
+5. Periodically probing the connectivity healthiness of the tunnel.
+
+#### Expected Usage
+
+Basically all the client library supporting HTTP proxy are expected to use 
+the proxy addons w/o additional work. As a matter of fact, most of those 
+popular HTTP client libraries only supports unsecured HTTP proxy instead of 
+mTLS-secured HTTPS proxy, and neither golang SDK nor JDK has builtin support 
+for mTLS proxy. So at the cost of securing the proxy server further, we will 
+need to adapt HTTPS proxy by our own effort. The good news is that the
+apiserver-network-proxy provided packages to talk over mTLS proxy. And also 
+there are a few https client binaries that natively supports mTLS proxy e.g.
+`curl` in the latest versions as is shown below:
+
+```bash
+curl -v -p \
+     --proxy-key certs/frontend/private/proxy-client.key \
+     --proxy-cert certs/frontend/issued/proxy-client.crt \
+     --proxy-cacert certs/frontend/issued/ca.crt \
+     --proxy-cert-type PEM \
+     -x https://127.0.0.1:8090 \   # the proxy-server's address
+     http://localhost:8000/success # any target L4 address
+````
+
+
+#### API Spec
+
+ManagedClusterAddon:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddon
+metadata:
+ name: apiserver-network-proxy
+spec:
+ addonMeta:
+   displayName: apiserver-network-proxy
+   description: "https://github.com/kubernetes-sigs/apiserver-network-proxy"
+ addonConfiguration:
+   crd: proxy.addon.open-cluster-management.io
+   name: proxy-config
+```
+
+ProxyConfiguration:
+
+```yaml
+apiVersion: proxy.addon.open-cluster-management.io/v1alpha1
+kind: ProxyConfiguration
+metadata:
+ name: apiserver-network-proxy
+spec:
+  server:
+    replicas: 2
+    protocol: [websocket|grpc]
+    tunnel-port: 8091
+    proxy-port: 8090
+  agent:
+    replicas: 2
+    server-address: 4.4.4.4
+```
+
```

Comment:
We also need the [`ManagedClusterAddOn `](https://github.com/open-cluster-management-io/api/blob/main/addon/v1alpha1/types_managedclusteraddon.go) to trigger the anp installation on each managed cluster

Reply:
And I think we may consider to add some status in the config/addon apis to check the anp and its connection status

Reply:
@skeeey i just added a few details about the config coordinates spec, let me know if i missed anything

Reply:
I think it is clear for me